src/main/java/sg/ncl/DataController.java

```diff
@@ -61,6 +61,7 @@ public String data(Model model) {
             datasetManager.addDataset(dataset);
         }
 
+        model.addAttribute("categories", getDataCategories());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "categories" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)